### PR TITLE
refactor(ui5-table): rename busy, busyDelay and mode properties, TableMode enum and its values

### PR DESCRIPTION
**Pull request by niyap**
Renames the `busy`, `busyDelay` and `mode` properties of `ui5-table`.
Also renames the `TableMode` enum to `TableSelectionMode` as well as its values from `SingleSelect` to `Single` and from `MultiSelect` to `Multi`.

If you have previously used the `busy`, `busyDelay` and `mode` properties and the TableMode enum:
```html
<ui5-table busy busy-delay="0" mode="SingleSelect"></ui5-table>
```
```js
import TableMode from "@ui5/webcomponents/dist/types/TableMode.js";
```

Now use `loading`, `loadingDelay` and `selectionMode` as well as TableSelectionMode enum instead:
```html
<ui5-table loading loading-delay="0" selectionMode="Single"></ui5-table>
```
```js
import TableSelectionMode from "@ui5/webcomponents/dist/types/TableSelectionMode.js";
```

BREAKING CHANGE: The `busy` and `mode` properties as well as the `TableMode` enum and its values have been renamed.

Related to: https://github.com/SAP/ui5-webcomponents/issues/8461

### CODE REVIEW

File: packages/main/src/Table.ts
Comment (hristop) [inline packages/main/src/Table.ts:309]: Should busy be changed to loading here as well?

**Review by hristop — COMMENTED** [review]
Should we change the busyDelay to loadingDelay as well? 
https://github.com/SAP/ui5-webcomponents/pull/8779/files#diff-c1eacd43766827fdb6c05695d82eb71acd650e1285bfc0287fb7bba82a89cc3cR323

**Review by hristop — DISMISSED** [review]

**Review by hristop — COMMENTED** [review]
The core team would like to discuss this before we merge it.

**Comment by ilhan007** [comment]
Hi colleagues, @hristop, @niyap, thank you for preparing the PR, but I think it's besto to close it - introducing breaking changes on a component that will be phased out in future and currently being moved to another package is not valuable and unnecessary. https://github.com/SAP/ui5-webcomponents/pull/8849

**Pull request closed without merging**
